agent/src/main/java/com/criteo/hadoop/garmadon/agent/tracers/hadoop/resourcemanager/RMContextImplEventRunnable.java

```diff
@@ -49,23 +61,23 @@ public void sendAppEvent(ApplicationId applicationId, RMApp rmApp) {
                 .withApplicationName(rmApp.getName())
                 .withFramework(rmApp.getApplicationType().toUpperCase());
 
-            ResourceManagerEventProtos.ApplicationEvent.Builder eventBuilder = ResourceManagerEventProtos.ApplicationEvent.newBuilder()
+            ApplicationEvent.Builder eventBuilder = ApplicationEvent.newBuilder()
                 .setState(rmApp.getState().name())
                 .setQueue(rmApp.getQueue());
 
             rmApp.getApplicationTags().stream()
-                    .filter(tag -> !tag.contains(":"))
+                    .filter(tag -> YARN_TAGS_TO_EXTRACT.stream().noneMatch(tag::startsWith))
                     .forEach(eventBuilder::addYarnTags);
 
             rmApp.getApplicationTags().stream()
-                    .filter(tag -> tag.contains(":"))
+                    .filter(tag -> YARN_TAGS_TO_EXTRACT.stream().anyMatch(tag::startsWith))
                     .map(tag -> {
                         int idx = tag.indexOf(':');
                         String key = tag.substring(0, idx);
```

Comment:
will failed if ":" is not found on tag string as idx will be -1

Reply:
Done